src/Controls/src/Core/Xaml/Diagnostics/VisualDiagnostics.cs

```diff
@@ -27,7 +27,7 @@ internal static void OnChildAdded(Element parent, Element child)
 			if (child is null)
 				return;
 
-			var index = parent?.AllChildren.IndexOf(child) ?? -1;
+			var index = (parent as IVisualTreeElement)?.GetVisualChildren().IndexOf(child) ?? -1;
```

Comment:
@mattleibow @drasticactions ?

Reply:
Looks good to me. As long as the tests are good, I'm good. 👍